csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -80,15 +79,20 @@ public int GetFieldIndex(string name, StringComparer comparer)
 
         public int GetFieldIndex(string name, IEqualityComparer<string> comparer = default)
         {
-            comparer ??= StringComparer.CurrentCulture;
+            if (comparer == null || comparer.Equals(StringComparer.CurrentCulture))
+            {
+                return _fieldsIndexLookup[name].First();
+            }
 
-            for (int i = 0; i < _fieldsList.Count; i++)
+            for (var i = 0; i < _fieldsList.Count; ++i)
```

Comment:
Please don't make unnecessary stylistic changes.

Reply:
Sure, will undo this.